src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1099,55 +1096,48 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
 
                 SetRequestContent(request, formData);
             }
-            // coerce body into a usable form
             else if (Body is not null)
             {
+                // Coerce body into a usable form
                 object content = Body;
 
-                // make sure we're using the base object of the body, not the PSObject wrapper
-                PSObject psBody = Body as PSObject;
-                if (psBody is not null)
+                // Make sure we're using the base object of the body, not the PSObject wrapper
+                if (Body is PSObject psBody)
                 {
                     content = psBody.BaseObject;
                 }
 
-                if (content is FormObject form)
-                {
-                    SetRequestContent(request, form.Fields);
-                }
-                else if (content is IDictionary dictionary && request.Method != HttpMethod.Get)
```

Comment:
`content is IDictionary dictionary && request.Method != HttpMethod.Get` is this the correct logic?

Reply:
I don't know. We need more investigations, perhaps we have tests for the code.

Reply:
I think the logic is wrong. I'll open an issue to discuss